File: examples/collections/collections.kt
enum class Food {
  vegetable,
  fruit,
  fastfood
}

val farm = hashMapOf(
  Food.vegetable to setOf("🍅", "🌽"),
  Food.fruit to setOf("🍌", "🍏")
)

fun main() {
  val vegetables = farm[Food.vegetable]
  if (vegetables != null) {
    if (vegetables.contains("🍅")) {
      println("We have tomatoes!")
    }
  }
}

File: examples/collections/collections.ts
enum Food {
  vegetable,
  fruit,
  fastfood
}

const farm = new Map([
  [Food.vegetable, new Set(["🍅", "🌽"])],
  [Food.fruit, new Set(["🍌", "🍏"])]
]);

const vegetables = farm.get(Food.vegetable);
if (vegetables) {
  if (vegetables.has("🍅")) {
    console.log("We have tomatoes!");
  }
}
